lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2474,7 +2474,15 @@ public function setOriginalEntityProperty($oid, $property, $value)
      */
     public function getEntityIdentifier($entity)
     {
-        return $this->entityIdentifiers[spl_object_id($entity)];
+        $id = spl_object_id($entity);
```

Comment:
This can be optimized to:
```suggestion
        return $this->entityIdentifiers[spl_object_id($entity)] ?? EntityNotFoundException::noIdentifierFound(get_class($entity));
```
This keeps the number of executed opcodes lower in this hot path.